src/axom/inlet/docs/sphinx/simple_types.rst

```diff
@@ -0,0 +1,144 @@
+   :language: lua
+
+
+Defining And Storing
+--------------------
+
+This example shows how to add the four simple field types with descriptions to the
+input file schema and add their values, if present in the input file, to the Sidre
+DataStore to be accessed later.
+
+.. literalinclude:: ../../examples/fields.cpp
+   :start-after: _inlet_simple_types_fields_add_start
+   :end-before: _inlet_simple_types_fields_add_end
+   :language: C++
+
+You can also add default values to Fields to fall back to if they are not defined
+in your input file. The last added Field was intentionally not present in the input file.  Not all
+fields need to be present, unless they are marked required, like ``a_simple_double``.
+
+Accessing
+---------
+
+Accessing field values stored in Inlet can be accessed via their name with the ``[]`` operator.
+
+.. literalinclude:: ../../examples/fields.cpp
+   :start-after: _inlet_simple_types_fields_access_start
+   :end-before: _inlet_simple_types_fields_access_end
+   :language: C++
+
+Here is an example of how to check if the input file contained an optional field:
+
+.. literalinclude:: ../../examples/fields.cpp
+   :start-after: _inlet_simple_types_fields_contains_start
+   :end-before: _inlet_simple_types_fields_contains_end
+   :language: C++
+
+
+******
+Tables
+******
+
+Tables help with grouping associated data together into a single container. Tables can
+contain multiple individually named Fields, multiple sub-Tables, as well as a single
+array or a single dictionary.
+
+In this example, we will be using the following part of an input file:
+
+.. literalinclude:: ../../examples/tables.cpp
+   :start-after: _inlet_simple_types_tables_input_start
+   :end-before: _inlet_simple_types_tables_input_end
+   :language: lua
+
+Defining And Storing
+--------------------
+
+This example shows how to add a Table with a nested Table to the
+input file schema and add the underlying field values to the Sidre
+DataStore to be accessed later.
+
+.. literalinclude:: ../../examples/tables.cpp
+   :start-after: _inlet_simple_types_tables_add_start
+   :end-before: _inlet_simple_types_tables_add_end
+   :language: C++
+
+This example also shows that the ``color`` Field that was not given in the
+input file but used the default value that was specified in the schema.
+
+.. note:: Although nested Tables are added with the ``addStruct`` method,
+   Inlet does not require any correspondence between nested structures defined
+   in the schema via ``addStruct`` and C++ ``struct`` s.
+   See :ref:`Advanced Types <inlet_advanced_types_label>` for more info.
+
+Accessing
+---------
+
+Accessing field values stored inside a table can be accessed via their name with the ``[]`` operator.
+They can be accessed from the Inlet class instance with their fully qualified name or you
+can get the Table instance first, then access it with the relative name.
+
+.. literalinclude:: ../../examples/tables.cpp
```

Comment:
Is there a way to extract a `Car` instance via `get<Car>`?  
(I'm guessing this is described later.)

Reply:
You would have to do the FromInlet instantiation.  I was trying to keep his example as simple as possible. Maybe in a follow up example (or alternate in this file) we could show how to do that as an "advanced" car example.